Sources/TensorFlow/Operators/Basic.swift

```diff
@@ -162,7 +188,23 @@ public extension Tensor {
     @inlinable
     @differentiable(wrt: self, vjp: _vjpReshaped(toShape:) where Scalar: TensorFlowFloatingPoint)
     func reshaped(toShape newShape: Tensor<Int32>) -> Tensor {
-        _Raw.reshape(self, shape: newShape)
+        let totalNegative = newShape.scalars.filter({$0 == -1}).count
```

Comment:
Unfortunately, these will also inhibit optimizations. :-( That said, I think it'd be really good to catch issues as early as possible and have good quality error messages. I believe we can eventually get good error messages in the fully dynamic cases as we upgrade our underlying runtime.

How about the following short-term compromise between usability and performance of: adding the preconditions to the `TensorShape` overload of `reshaped` above, where we can eagerly check the shape errors. WDYT?

Reply:
That is actually something I though. While I was testing the methods I found the error messages were quite understandable. I think the issue was created because there are some methods that just break the applications without any error. May be we can modify #517 to find the methods that crash without human readable error error and trace?

Reply:
I checked with @pschuh and @asuhan and the new tracing system that's coming soon, and the performance concerns I was worried about are not important. Looking forward to getting this PR merged!